### ISSUE TITLE
Implement Application Template feature

### ISSUE THREAD

**Issue by PK85** [kyma-incubator/compass#428]
**Description**
New Application Template feature allows compass users to:
- ~manage application templates owned by integration system~ - out of scope
  - ~list AT for given integration system~
  - ~prevent deletion of the IntegrationSystem when some AT are assigned to~
- manage templates from applications + placeholders
- define and use input parameters that need to be provided when application is created from template
- (in the future, blocked by proposal: https://github.com/kyma-incubator/compass/issues/415) add new PairingAuthSystem related to given IntegrationSystem. Then created application from AT will use generic getAuth() method that wraps a call to the PairingAuthSystem.

**AC**
- implement Application Templates (based on approved proposal: https://github.com/kyma-incubator/compass/issues/339)
- create E2E tests for two main scenarios mentioned in the description:
    - manage application templates owned by integration system
    - manage templates from applications + placeholders
- add final documentation about integration systems to kyma and sync kyma and compass docs
- add final documentation about application templates to kyma and sync kyma and compass docs
- every sub-feature must be well documented and tested (in case of dropping features, list them in a comment and discuss with PO)

**Comment by pkosiec** [comment]
The following ACs are out of scope:
- manage application templates owned by integration system
  - list AT for given integration system
  - prevent deletion of the IntegrationSystem when some AT are assigned to

We could implement listing ApplicationTemplates with labels, but it would require us to implement global Labels and LabelDefinitions, as well as additional mutations:
   - `setApplicationTemplateLabel`
   - `deleteApplicationTemplateLabel`
   - `createGlobalLabelDefinition` -> or additional parameter for `createLabelDefinition`. But then we need custom logic for scopes validation
   - `updateLabelDefinition` -> or additional parameter for `createLabelDefinition`. But then we need custom logic for scopes validation
   - `deleteGlobalLabelDefinition` -> or additional parameter for `createLabelDefinition`. But then we need custom logic for scopes validation
